pyracf/common/security_admin.py

```diff
@@ -157,32 +179,59 @@ def _make_request(
+        )
         if self._generate_requests_only:
-            request_xml = self.__logger.redact_request_xml(
-                security_request.dump_request_xml(encoding="utf-8"),
-                secret_traits=self.__secret_traits,
-            )
             self.__clear_state(security_request)
             return request_xml
         result_xml = self.__logger.redact_result_xml(
             self.__irrsmo00.call_racf(
-                security_request.dump_request_xml(), irrsmo00_precheck
+                security_request.dump_request_xml(),
+                irrsmo00_precheck,
+                self.__running_userid,
             ),
             self.__secret_traits,
         )
         self.__clear_state(security_request)
+        if isinstance(result_xml, list):
+            # When IRRSMO00 encounters some errors, it returns no XML response string.
+            # When this happens, the C code instead surfaces the return and reason
+            # codes which causes a DownstreamFatalError to be raised.
+            raise DownstreamFatalError(
+                return_reason_codes=result_xml,
+                request_xml=request_xml,
+                run_as_userid=self.get_running_userid(),
+            )
         if self.__debug:
             # No need to redact anything here since the raw result XML
             # already has secrets redacted when it is built.
             self.__logger.log_xml("Result XML", result_xml)
-        results = SecurityResult(result_xml)
+        results = SecurityResult(result_xml, self.get_running_userid())
         if self.__debug:
             # No need to redact anything here since the result dictionary
             # already has secrets redacted when it is built.
             self.__logger.log_dictionary(
                 "Result Dictionary", results.get_result_dictionary()
             )
         result_dictionary = results.get_result_dictionary()
+        if result_dictionary["securityResult"]["returnCode"] > 4:
+            # All return codes greater than 4 are indicative of an issue with
+            # IRRSMO00 that would stop a command image from being generated.
+            # The user should interogate the result dictionary attached to the
+            # SecurityRequestError as well as the return and reason codes to
+            # resolve the problem.
+            raise DownstreamFatalError(
+                [
+                    8,
+                    result_dictionary["securityResult"]["returnCode"],
+                    result_dictionary["securityResult"]["reasonCode"],
+                ],
+                request_xml,
+                self.get_running_userid(),
+                result_dictionary,
+            )
         if result_dictionary["securityResult"]["returnCode"] != 0:
             # All non-zero return codes should cause a SecurityRequestError to be raised.
             # Even if a return code of 4 is not indicative of a problem, it it is
```

Comment:
`it` is repeated. I think this was my mistake.

Reply:
yes, fixed.